docker_test.go

```diff
@@ -46,6 +49,62 @@ func init() {
 	}
 }
 
+func TestWithinContainerStage1(t *testing.T) {
+	if providerType != ProviderDocker {
+		t.Skip("This is a docker-specific test.")
+	}
+	dir, err := os.Getwd()
+	require.NoError(t, err)
+	req := ContainerRequest{
+		Image:      golangImage,
+		WorkingDir: dir,
+		HostConfigModifier: func(hc *container.HostConfig) {
+			hc.Mounts = append(hc.Mounts, mount.Mount{
+				Type:   "bind",
+				Source: dir,
+				Target: dir,
+			}, mount.Mount{
+				Type:   "bind",
+				Source: "/var/run",
+				Target: "/var/run",
+			})
+		},
+		Entrypoint: []string{"sleep", "100"},
+	}
+	container, err := GenericContainer(context.Background(), GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	})
+	require.NoError(t, err)
+	// The timeout is necessary because when TestWithinContainersStage2 fails, it hangs forever instead of erroring
+	exitCode, outputReader, err := container.Exec(context.Background(), []string{"go", "test", "-v", "-run", "^TestWithinContainerStage2$", "-timeout", "10s"}, exec.Multiplexed())
+	require.NoError(t, err)
+
+	outputStr, err := io.ReadAll(outputReader)
+	if err != nil {
+		t.Fatalf("failed to read output: %s", err)
+	}
+	t.Logf("%s", outputStr)
+
+	if exitCode != 0 {
+		t.Fatalf("exit code %d, output: %s", exitCode, outputStr)
+	}
```

Comment:
suggestion: use require.Zero
```suggestion
	require.Zerof(t, exitCode, "output: %s", outputStr)
```